### PR TITLE
CASMINST-3128 Cleanup of the update docs

### PR DESCRIPTION
**Pull request by cdelatte-hpe**

### CODE REVIEW

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Stage 0 - Prerequisites and Preflight Checks
+
+> NOTE: CSM-0.9.4 or CSM-0.9.5 is the version of CSM required in order to upgrade to CSM-1.0.0 (available with Shasta v1.5).
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:3]: Bold instances of NOTE: for consistency with rest of docs.

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+>
+> NOTE: Installed CSM versions may be listed from the product catalog using the following command. This will sort a semantic version without a hyphenated suffix after the same semantic version with a hyphenated suffix, e.g. 1.0.0 > 1.0.0-beta.19.
+>
+
+The following command can be used to check the CSM version on the system:
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:8]: Change to "Use the following command to check the CSM version on the system:"

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+3. Update the `site-init` secret:
+
+   ```bash
+   ncn-m001# kubectl delete secret -n loftsman site-init
+   ncn-m001# kubectl create secret -n loftsman generic site-init --from-file=customizations.yaml
+   ```
+
+4. If using an external Git repository for managing customizations ([as recommended](../../install/prepare_site_init.md#version-control-site-init-files)),
+   clone a local working tree and commit appropriate changes to `customizations.yaml`,
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:60]: Change comma to period after customization.yaml.

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+   e.g.:
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:61]: Change to "For example:"

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+1. Run check script:
+
+    * Internet Connected
+
+        ```bash
+        ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/prerequisites.sh --csm-version [CSM_RELEASE] --endpoint [ENDPOINT]
+        ```
+
+        **NOTE** ENDPOINT is optional for internal use. It is pointing to internal arti by default
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:88]: Missing period.

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    * Air Gapped
+
+        ```bash
+        ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/prerequisites.sh --csm-version [CSM_RELEASE] --tarball-file [PATH_TO_CSM_TARBALL_FILE]
+        ```
+
+**`IMPORTANT:`** If any errors are encountered then potential fixs should be displayed where the error occurred. `IF` the upgrade prequesites.sh script does not provide guidance, then open a support ticket for guidance before proceeding.
+
+2. The script also runs the goss tests on the initial stable node (typically `ncn-m001`) where the latest version of CSM has been installed. Make sure the goss test pass before continue.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:98]: - This step should start with the action item. Change to something like:
"Ensure the goss tests pass before continuing. The script runs the goss tests on the initial stable node (typically `ncn-m001`) where the latest version of CSM has been installed." 

- Do we need to provide guidance on how to check if the tests pass?

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+## Stage 0.4 - Backup VCS Data
+
+To prevent any possibility of losing configuration data, backup the VCS data and store it in a safe location.  See [Version_Control_Service_VCS.md](../../operations/configuration_management/Version_Control_Service_VCS.md#backup-and-restore-data) for these procedures.
+
+**`IMPORTANT:`** As Part of this stage, only perform the backup steps.  `Only execute the restore if necessary.`  You will not always restore after a backup as we callout the backup procedure as a precautionary step.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:104]: "Part" should not be capitalized.

File: upgrade/1.0/Stage_0_Prerequisites.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+Once the above steps have been completed please proceed to [Stage 1](Stage_1.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_0_Prerequisites.md:106]: Add comma after "completed". Missing period at end of sentence.

File: upgrade/1.0/Stage_1.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+## Stage 1.1
+
+Run:
+
+```bash
+ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/ncn-upgrade-ceph-initial.sh ncn-s001
+```
+
+> NOTE: Run the script once each for all storage nodes. Follow output of the script carefully. The script will pause for manual interaction
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:11]: Bold NOTE: for consistency.

File: upgrade/1.0/Stage_1.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
+# Stage 1 - Ceph upgrade from Nautilus (14.2.x) to Octopus (15.2.x)
+
+## Stage 1.1
+
+Run:
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:5]: Add a little more content. Maybe, "Run the ncn-upgrade-ceph-initial.sh script:"

File: upgrade/1.0/Stage_1.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+```bash
+ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/ncn-upgrade-ceph-initial.sh ncn-s001
+```
+
+> NOTE: Run the script once each for all storage nodes. Follow output of the script carefully. The script will pause for manual interaction
+
+## Stage 1.2
+
+**`IMPORTANT:`** We scale down the conman deployments during stage 1.2 (this stage), so all console sessions will be down for this portion of the upgrade
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:15]: - conman -> ConMan

- Missing period.

File: upgrade/1.0/Stage_1.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+/etc/cray/ceph/mons_upgraded
+/etc/cray/ceph/mgrs_upgraded
+/etc/cray/ceph/keys_distributed
+/etc/cray/ceph/converted_to_orch
+/etc/cray/ceph/osds_upgraded
+/etc/cray/ceph/mds_upgraded
+/etc/cray/ceph/rgws_upgraded
+```
+
+**`NOTE:`** You can delete all these files and rerun or you can just delete any files from your last step. You may end up with checkpoint files if the upgrade was exited out of by the user. But if you know you exited out of OSDs and it was not a clean exit, then you would only need remove `osd_upgraded`, `mds_upgraded` and `rgw_upgraded`.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:34]: - Comma needed after rerun

- Comma needed after `mds_upgraded`

File: upgrade/1.0/Stage_1.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Start the Ceph upgrade
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:36]: missing punctuation

File: upgrade/1.0/Stage_1.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+   > * Both methods are dependent on the master nodes and storage nodes 001/2/3 have a ceph.client.admin.keyring and/or a ceph.conf file    (cephadm will not require the ceph.conf).
+   > * When you continue with Stage 2, you may have issues running your Ceph commands.
+   >   * If you are experiencing this, please double check that you restored your /etc/ceph directory from your tar backup.
+   > * Any deployments that are backed by a cephfs PVC will be unavailable during this stage of the upgrade. These deployments will be    scaled down and back up automatically. This includes **(but can vary by deployment)**: `nexus`, `cray-ipxe`, `cray-tftp`, `cray-ims`,    `cray-console-operator`, and `cray-cfs-api-db`. To view the complete list for the system being upgraded, run the following script to list    them:
+    >>
+    >>   ```bash
+    >>   ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/list-cephfs-clients.sh
+    >>   ```
+
+2. Verify that you conman is running
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:56]: - conman -> ConMan
- missing punctuation
- remove "you"

File: upgrade/1.0/Stage_1.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    cray-console-data-postgres-1                                   3/3     Running            0          4d23h
+    cray-console-data-postgres-2                                   3/3     Running            0          5d
+    cray-console-data-wait-for-postgres-5-jrsq4                    0/2     Completed          0          3d22h
+    cray-console-node-0                                            3/3     Running            0          5d
+    cray-console-node-1                                            3/3     Running            0          5h15m
+    cray-console-operator-c4748d6b4-vvpn7                          2/2     Running            0          4d23h
+    csm-config-import-1.0.0-beta.46-5t7kx                          0/3     Completed          0          4d
+    ```
+
+**`NOTE:`** if conman is not running please see [establishing conman console connections](operations/../../../operations/conman/Establish_a_Serial_Connection_to_NCNs.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:72]: - Capitalize "if"
- conman -> ConMan
- missing period

File: upgrade/1.0/Stage_1.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+Once the `Stage 1` upgrade is complete please proceed to [Stage 2](Stage_2.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_1.md:74]: - comma needed after "complete"
- missing period

File: upgrade/1.0/Stage_2.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
+# Stage 2 - Ceph image upgrade
+
+For each storage node in the cluster, start by following the steps:
+
+```bash
+ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/ncn-upgrade-ceph-nodes.sh ncn-s001
+```
+
+> NOTE: Run the script once each for all storage nodes. Follow output of the script carefully. The script will pause for manual interaction
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:9]: Bold NOTE:

File: upgrade/1.0/Stage_2.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+> Note that these steps should be performed on one storage node at a time.
+
+After the last storage node has been rebooted you will need to deploy node-exporter and alertmanager
+
+**NOTE:** This process will need to run on a node running ceph-mon, which in most cases will be ncn-s00(1/2/3)
+**NOTE:** You may need to reset the root password for each node after it is rebooted
+
+
+1. Deploy node-exporter and alertmanager
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:18]: missing punctuation

File: upgrade/1.0/Stage_2.md [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+```bash
+ncn-m001# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/ncn-upgrade-ceph-nodes.sh ncn-s001
+```
+
+> NOTE: Run the script once each for all storage nodes. Follow output of the script carefully. The script will pause for manual interaction
+> Note that these steps should be performed on one storage node at a time.
+
+After the last storage node has been rebooted you will need to deploy node-exporter and alertmanager
+
+**NOTE:** This process will need to run on a node running ceph-mon, which in most cases will be ncn-s00(1/2/3)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:14]: missing punctuation

File: upgrade/1.0/Stage_2.md [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+**NOTE:** You may need to reset the root password for each node after it is rebooted
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:15]: missing punctuation

File: upgrade/1.0/Stage_2.md [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+
+    ```bash
+    ncn-s00(1/2/3)# ceph orch apply node-exporter
+    Scheduled node-exporter update...
+
+    ncn-s00(1/2/3)# ceph orch apply alertmanager
+    Scheduled alertmanager update...
+    ```
+
+2. Verify node-exporter and alertmanager are running
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:28]: missing punctuation

File: upgrade/1.0/Stage_2.md [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+  **IMPORTANT:** There should be a node-exporter container per ceph node and a single alertmanager container for the cluster.
+
+1. Update BSS to ensure the ceph images are loaded if a node is rebuilt.
+
+    ```bash
+    . /usr/share/doc/csm/upgrade/1.0/scripts/ceph/lib/update_bss_metadata.sh
+    update_bss_storage
+    ```
+
+  Once `Stage 2` is completed and all the ceph node have been rebooted into the new image, then please proceed to [Stage 3](Stage_3.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:51]: - missing punctuation
- ceph -> Ceph

File: upgrade/1.0/Stage_2.md [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+     node-exporter.ncn-s003  ncn-s003  running (57m)  3m ago     67m  0.18.1   registry.local/prometheus/node-exporter:v0.18.1  e5a616e4b9cf       1078d9e555e4
+
+     ncn-s00(1/2/3)# ceph orch ps --daemon_type alertmanager
+     NAME                   HOST      STATUS         REFRESHED  AGE  VERSION  IMAGE NAME                                      IMAGE ID           CONTAINER ID
+     alertmanager.ncn-s001  ncn-s001  running (66m)  3m ago     68m  0.20.0   registry.local/prometheus/alertmanager:v0.20.0  0881eb8f169f       775aa53f938f
+     ```
+
+  **IMPORTANT:** There should be a node-exporter container per ceph node and a single alertmanager container for the cluster.
+
+1. Update BSS to ensure the ceph images are loaded if a node is rebuilt.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:44]: - First instance of BSS. Spell out Boot Script Service (BSS).
- ceph -> Ceph

File: upgrade/1.0/Stage_2.md [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+     node-exporter.ncn-s001  ncn-s001  running (57m)  3m ago     67m  0.18.1   docker.io/prom/node-exporter:v0.18.1             e5a616e4b9cf       3465eade21da
+     node-exporter.ncn-s002  ncn-s002  running (57m)  3m ago     67m  0.18.1   registry.local/prometheus/node-exporter:v0.18.1  e5a616e4b9cf       7ed9b6cc9991
+     node-exporter.ncn-s003  ncn-s003  running (57m)  3m ago     67m  0.18.1   registry.local/prometheus/node-exporter:v0.18.1  e5a616e4b9cf       1078d9e555e4
+
+     ncn-s00(1/2/3)# ceph orch ps --daemon_type alertmanager
+     NAME                   HOST      STATUS         REFRESHED  AGE  VERSION  IMAGE NAME                                      IMAGE ID           CONTAINER ID
+     alertmanager.ncn-s001  ncn-s001  running (66m)  3m ago     68m  0.20.0   registry.local/prometheus/alertmanager:v0.20.0  0881eb8f169f       775aa53f938f
+     ```
+
+  **IMPORTANT:** There should be a node-exporter container per ceph node and a single alertmanager container for the cluster.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_2.md:42]: ceph -> Ceph

File: upgrade/1.0/Stage_3.md [hunk 25]
```diff
@@ -0,0 +1,94 @@
+# Stage 3 - Kubernetes Upgrade from 1.18.6 to 1.19.9
+
+> NOTE: During the CSM-0.9 install the LiveCD containing the initial install files for this system should have been unmounted from the master node when rebooting into the Kubernetes cluster. The scripts run in this section will also attempt to unmount/eject it if found to ensure the USB stick does not get erased.
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:3]: Bold NOTE: in all instances below

File: upgrade/1.0/Stage_3.md [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+## Stage 3.4
+
+Run the following command to complete the Kubernetes upgrade _(this will restart several pods on each master to their new docker containers)_:
+
+```bash
+ncn-m002# export PDSH_SSH_ARGS_APPEND="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ncn-m002# pdsh -b -S -w $(grep -oP 'ncn-m\d+' /etc/hosts | sort -u |  tr -t '\n' ',') 'kubeadm upgrade apply v1.19.9 -y'
+```
+
+> **`NOTE`**: kubelet has been upgraded already so you can ignore the warning to upgrade kubelet
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:81]: missing period.

File: upgrade/1.0/Stage_3.md [hunk 27]
```diff
@@ -0,0 +1,94 @@
...
+
+Run `csm-service-upgrade.sh` to deploy upgraded CSM applications and services:
+
+```bash
+ncn-m002# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/csm-service-upgrade.sh
+```
+
+**`IMPORTANT`:** This script will re-try up to three times if failures are encountered -- but if the script seems to hang for thirty minutes or longer without progressing, the administrator should interrupt the script (CTRL-C) and re-run it.
+
+Once `Stage 3` is completed and all kubernetes nodes have been rebooted into the new image then please proceed to [Stage 4](Stage_4.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:94]: - Missing period
- Add comma after image
- kubernetes -> Kubernetes

File: upgrade/1.0/Stage_4.md [hunk 28]
```diff
@@ -0,0 +1,9 @@
+# Stage 4 - CSM Service Upgrades
+
+Run `csm-service-upgrade.sh` to deploy upgraded CSM applications and services:
+
+```bash
+ncn-m002# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/csm-service-upgrade.sh
+```
+
+Once the `Stage 4` service upgrade is complete then please proceed to [Stage 5](Stage_5.md)
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_4.md:9]: Change to:

Once the `Stage 4` service upgrade is complete, proceed to [Stage 5](Stage_5.md).

File: upgrade/1.0/Stage_3.md [hunk 29]
```diff
@@ -0,0 +1,94 @@
...
+`Option 1` - Internet Connected Environment
+
+Install document RPM package:
+
+```bash
+ncn-m002# wget https://storage.googleapis.com/csm-release-public/shasta-1.5/docs-csm/docs-csm-latest.noarch.rpm
+ncn-m002# rpm -Uvh docs-csm-latest.noarch.rpm
+```
+
+Run:
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:42]: Add more context on what is being run or what is happening here.

File: upgrade/1.0/Stage_3.md [hunk 30]
```diff
@@ -0,0 +1,94 @@
...
+
+`Option 2` - Air Gapped Environment
+
+Install document RPM package:
+
+```bash
+ncn-m002# rpm -Uvh [PATH_TO_docs-csm-*.noarch.rpm]
+```
+
+Run:
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:58]: Add more context on what is being run or what is happening here.

File: upgrade/1.0/Stage_3.md [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+ncn-m002# rpm -Uvh docs-csm-latest.noarch.rpm
+```
+
+Run:
+
+```bash
+ncn-m002# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/prerequisites.sh --csm-version [CSM_RELEASE] --endpoint [ENDPOINT]
+```
+
+**NOTE** ENDPOINT is optional for internal use. It is pointing to internal arti by default
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:48]: missing period

File: upgrade/1.0/Stage_3.md [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+ncn-m002# rpm -Uvh [PATH_TO_docs-csm-*.noarch.rpm]
+```
+
+Run:
+
+```bash
+ncn-m002# /usr/share/doc/csm/upgrade/1.0/scripts/upgrade/prerequisites.sh --csm-version [CSM_RELEASE] --tarball-file [PATH_TO_CSM_TARBALL_FILE]
+```
+
+> NOTE: Follow output of above script carefully. The script will pause for manual interaction
```
Comment (nrockershousen) [inline upgrade/1.0/Stage_3.md:64]: missing period

**Review by phalseth-hpe — APPROVED** [review]

**Review by heemstra — APPROVED** [review]

**Pull request merged**